### PR TITLE
HARP-5189: Add a simple SubdivisionModifier.

### PR DESCRIPTION
**Pull request by robertoraggi**
The SubdivisionModifier is used to increase the number of
vertices of polygons meant to be rendered on a sphere.

Signed-off-by: Roberto Raggi <roberto.raggi@here.com>

### CODE REVIEW

File: @here/harp-geometry/lib/SubdivisionModifier.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    /**
+     * Subdivides the faces of the given `THREE.Geometry`.
+     *
+     * @param geometry The `THREE.Geometry` to subdivide.
+     */
+    modify(geometry: Geometry): Geometry {
+        const { vertices, faces } = geometry;
+
+        const cache = new Map<string, number>();
```
Comment (ninok) [inline @here/harp-geometry/lib/SubdivisionModifier.ts]: What is this cache caching?

File: @here/harp-geometry/lib/SubdivisionModifier.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import {
+    Face3,
+    Geometry,
+} from "three";
+
+/**
+ * The `SubdivisionModifier` subdivides triangle mesh geometries positioned
+ * on the surface of a sphere centered at `(0, 0, 0)`.
+ */
+export class SubdivisionModifier {
```
Comment (ninok) [inline @here/harp-geometry/lib/SubdivisionModifier.ts]: Maybe rename to SphereSubdivisionModifier b/c for terrain we might use a different approach.

File: @here/harp-geometry/lib/SubdivisionModifier.ts (hunk 1 above)
Comment (zbigg) [reply @here/harp-geometry/lib/SubdivisionModifier.ts]: Middle vertices. 
`middleVertexCache` ?

File: @here/harp-geometry/lib/SubdivisionModifier.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            } else if (gamma === m) {
+                const d = middleVertex(face.c, face.a);
+                faces.push(new Face3(face.a, face.b, d));
+                faces.push(new Face3(d, face.b, face.c));
+            } else {
+                throw new Error("failed to subdivide the given geometry");
+            }
+        }
+
+        geometry.faces = subdividedFaces;
```
Comment (zbigg) [inline @here/harp-geometry/lib/SubdivisionModifier.ts]: Geometry may have some interdependencies between internal data.
Shouldn't we trigger one of these:
```
	elementsNeedUpdate;
	verticesNeedUpdate;
	this.uvsNeedUpdate =...
```
(https://threejs.org/docs/#api/en/core/Geometry)

File: @here/harp-geometry/lib/SubdivisionModifier.ts (hunk 2 above)
Comment (zbigg) [inline @here/harp-geometry/lib/SubdivisionModifier.ts]: I would question necessity to put this in separate class with only one method.

Our guidleines say:
> Use simplest tool for the job: function vs. object/namespace/module vs class.

This is plain "data->data" transformation, with no requirements for state (`angle` is hardly _state_).

My proposal `function subdivideGeometryToMatchSphere` ?

File: @here/harp-geometry/lib/SubdivisionModifier.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+    Face3,
+    Geometry,
+} from "three";
+
+/**
+ * The `SubdivisionModifier` subdivides triangle mesh geometries positioned
```
Comment (nzjony) [inline @here/harp-geometry/lib/SubdivisionModifier.ts]: Minor, should be [[SubdivisionModifier]]

File: @here/harp-geometry/lib/SubdivisionModifier.ts (hunk 2 above)
Comment (robertoraggi) [reply @here/harp-geometry/lib/SubdivisionModifier.ts]: I renamed it to `SphericalGeometrySubdivisionModifier`.

File: @here/harp-geometry/lib/SubdivisionModifier.ts (hunk 3 above)
Comment (robertoraggi) [reply @here/harp-geometry/lib/SubdivisionModifier.ts]: I updated the change to set vertices/elementsNeedUpdate. I also I added a paragraph in the documentation explaining that only the position and the faces of the geometry are modified, other common attributes of THREE.Geometry (like normals and UVs) are usually inferred by the vertex position so currently we don't need to preserve them, but if needed we can easily extend the code compute values for them.

File: @here/harp-geometry/lib/SubdivisionModifier.ts (hunk 2 above)
Comment (robertoraggi) [reply @here/harp-geometry/lib/SubdivisionModifier.ts]: the API is following the de-facto standard interface of three.js's modifiers.
https://github.com/mrdoob/three.js/tree/dev/examples/js/modifiers

File: @here/harp-geometry/lib/SubdivisionModifier.ts (hunk 4 above)
Comment (robertoraggi) [reply @here/harp-geometry/lib/SubdivisionModifier.ts]: done

File: @here/harp-geometry/lib/SubdivisionModifier.ts (hunk 1 above)
Comment (robertoraggi) [reply @here/harp-geometry/lib/SubdivisionModifier.ts]: I added some documentation explaining what is that we are caching,

**Pull request merged**